app/controllers/password_credentials_controller.rb

```diff
@@ -0,0 +1,32 @@
+class PasswordCredentialsController < ApplicationController
+  require_unauthenticated_access
+  before_action :ensure_manual_login_allowed
+
+  layout "public"
+
+  def edit
+    @user = find_signed_user(params[:token])
+  end
+
+  def update
+    user = find_signed_user(params[:token])
+
+    if user.password_credential&.update(update_params)
+      redirect_to login_path, notice: "Your password was reset succesfully. Please sign in."
```

Comment:
Would it be trivial to just log the user in at this point? If it's not trivial, let's not bother. But it's always a slightly nicer experience when the website simply lets you in right away since we're already trusting that you are the owner of this account.

I think it may simply be:

```
      login_as(person, credential: user.password_credential)
      redirect_to root_path, notice: "Your password was reset successfully."
```